lib/fog/libvirt/models/compute/server.rb

```diff
@@ -257,11 +257,128 @@ def cloud_init_volume_name
         private
         attr_accessor :volumes_path
 
-        # This retrieves the ip address of the mac address
+        # This tests the library version before redefining the address
+        # method for this instance to use a method compatible with
+        # earlier libvirt libraries, or uses the dhcp method from more
+        # recent releases.
+        def addresses(service_arg=service, options={})
+          addresses_method = self.method(:addresses_dhcp)
+          # check if ruby-libvirt was compiled against a new enough version
+          # that can use dhcp_leases, as otherwise it will not provide the
+          # method dhcp_leases on any of the network objects.
+          network = service.client.lookup_network_by_uuid(
+            service.networks.first.uuid)
```

Comment:
I suspect that part of this and the call below to `service.client.libversion` both need to be moved to the requests file in order to allow existing tests to pass and follow the current coding structure where the code that actually talks to libvirt is all in requests?